initfiles/componentfiles/configxml/@temp/wslogserviceespagent.xsl

```diff
@@ -23,100 +23,97 @@ xmlns:set="http://exslt.org/sets">
     <xsl:import href="esp_logging_transid.xsl"/>
 
     <xsl:template name="WsLogServiceESPAgent" type="DefaultLoggingAgent">
-        <xsl:param name="managerNode"/>
-        <xsl:for-each select="$managerNode/ESPLoggingAgent">
-            <xsl:variable name="agentName" select="@ESPLoggingAgent"/>
-            <xsl:variable name="agentNode" select="/Environment/Software/WsLogServiceESPAgent[@name=$agentName]"/>
-            <xsl:if test="not($agentNode)">
-                <xsl:message terminate="yes">An WsLogService ESP Logging Agent <xsl:value-of select="$agentName"/>  for <xsl:value-of select="$managerNode/@name"/> is undefined!</xsl:message>
+        <xsl:param name="agentName"/>
+        <xsl:param name="agentNode"/>
+        <xsl:if test="not($agentNode)">
+            <xsl:message terminate="yes">An WsLogService ESP Logging Agent <xsl:value-of select="$agentName"/> is undefined!</xsl:message>
+        </xsl:if>
+        <xsl:variable name="loggingServer" select="$agentNode/LoggingServer"/>
+        <xsl:if test="not($loggingServer)">
+            <xsl:message terminate="yes">ESP logging server is undefined for <xsl:value-of select="$agentName"/> </xsl:message>
+        </xsl:if>
+        <xsl:variable name="loggingServerUrl" select="$loggingServer/@Url"/>
+        <xsl:if test="string($loggingServerUrl) = ''">
+            <xsl:message terminate="yes">Logging Server URL is undefined!</xsl:message>
```

Comment:
Perhaps also mention the $agentName here too for easy reference which agent has the error.

Reply:
Good point. I will change it.